lib/bloc/location/location_bloc.dart

```diff
@@ -32,4 +39,37 @@ class LocationBloc extends Bloc<LocationEvent, LocationState> {
       emit(UnknownLocationErrorState());
     }
   }
+
+  Future<void> _requestPermissionLocationEvent(
+      Emitter<LocationState> emit) async {
+    final PermissionStatus permissionGranted =
+        await _locationService.requestPermission();
+    switch (permissionGranted) {
+      //permission granted
+      case PermissionStatus.granted:
+        final LocationDetails data =
+            await _locationService.getCurrentLocation();
+        final String dateDisplay =
+            DateFormat('dd-MMM-yyyy').format(DateTime.now());
+        emit(LocationReadyState(data, dateDisplay));
+        break;
+      //permission denied
+      case PermissionStatus.denied:
+        emit(LocationPermissionDeniedState());
+        break;
+      //permission deniedForever
+      case PermissionStatus.deniedForever:
+        emit(LocationPermissionDeniedState());
+        break;
+      default:
+    }
```

Comment:
it's better to update location_service's `Future<PermissionStatus> requestPermission` to return our own enum, and combine `PermissionStatus.denied` and `PermissionStatus.deniedForever` into 1 `denied` status.

that way, we do not have to import `package:location/location.dart` in our main bloc, and violating [`depend_on_referenced_packages`](https://dart-lang.github.io/linter/lints/depend_on_referenced_packages.html)